compiler/AST/checkAST.cpp

```diff
@@ -247,6 +247,7 @@ void checkPrimitives()
      case PRIM_CAPTURE_FN:
      case PRIM_CREATE_FN_TYPE:
      case PRIM_STRING_COPY:
+     case PRIM_MEMCPY:
```

Comment:
Maybe we could define memcpy to be not generic and to take c_void_ptr, and then we use a primitive to cast to c_void_ptr. Alternatively, we could make a special PRIM_MOVE somehow that codegen would know to generate as a memcpy. (The issue is that one or both pointers won't be aligned, which would cause core dump on some not-Intel architectures).